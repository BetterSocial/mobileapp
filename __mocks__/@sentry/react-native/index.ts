export const addBreadcrumb = jest.fn();
