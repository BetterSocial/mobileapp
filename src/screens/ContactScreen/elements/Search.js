import * as React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  TextInput,
  Animated,
} from 'react-native';

import MemoIcNewChat from '../../../assets/icons/ic_new_chat';
import MemoIc_search from '../../../assets/icons/Ic_search';
import StringConstant from '../../../utils/string/StringConstant';
import {COLORS, FONTS, SIZES} from '../../../utils/theme';

const Search = ({onPress, animatedValue}) => {
  return (
    <Animated.View style={styles.animatedViewContainer(animatedValue)}>
      <View style={styles.wrapperSearch}>
        <TextInput
          multiline={false}
          placeholder={'Search Users'}
          style={styles.input}
        />
        <View style={styles.wrapperIcon}>
          <MemoIc_search width={20} height={20} />
        </View>
      </View>
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base,
  },
  wrapperSearch: {
    flex: 1,
    backgroundColor: '#F5F5F5',
    marginLeft: 15,
    marginRight: 15,
    borderRadius: SIZES.radius,
    alignSelf: 'center',
  },
  wrapperButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginEnd: SIZES.base,
    paddingLeft: 8,
    paddingRight: 8,
    paddingTop: 9,
    paddingBottom: 9,
    // backgroundColor: 'red',
  },
  input: {
    marginLeft: 24,
    paddingStart: 16,
    lineHeight: 24,
  },
  wrapperIcon: {
    position: 'absolute',
    left: 8,
    top: 0,
    bottom: 0,
    justifyContent: 'center',
  },
  newPostText: {
    color: COLORS.holyTosca,
    marginRight: 11,
    ...FONTS.h4,
  },
  animatedViewContainer: (animatedValue) => ({
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base,
    marginTop: animatedValue,
    top: 0,
    left: 0,
    right: 0,
    zIndex: 10,
    padding: 7,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.gray1,
  }),
});

export default Search;
