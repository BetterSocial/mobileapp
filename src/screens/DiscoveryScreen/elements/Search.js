import * as React from 'react';
import {
  Animated,
  Pressable,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View
} from 'react-native';
import { useNavigation } from '@react-navigation/core'

import DiscoveryAction from '../../../context/actions/discoveryAction';
import DiscoveryRepo from '../../../service/discovery';
import GeneralComponentAction from '../../../context/actions/generalComponentAction';
import IconClear from '../../../assets/icon/IconClear';
import MemoIc_arrow_back_white from '../../../assets/arrow/Ic_arrow_back_white';
import MemoIc_pencil from '../../../assets/icons/Ic_pencil';
import MemoIc_search from '../../../assets/icons/Ic_search';
import StringConstant from '../../../utils/string/StringConstant';
import dimen from '../../../utils/dimen';
import {COLORS, FONTS, SIZES} from '../../../utils/theme';
import {Context} from '../../../context/Store';
import { colors } from '../../../utils/colors';
import {fonts} from '../../../utils/fonts';

let getDataTimeoutId;

const DiscoverySearch = ({onPress, animatedValue, showBackButton = false, onContainerClicked = () => {}}) => {
  const navigation = useNavigation()
  const [generalComponent, generalComponentDispatch] = React.useContext(Context).generalComponent
  const [discovery, discoveryDispatch] = React.useContext(Context).discovery
  const discoverySearchBarRef = React.useRef(null)
  const [isTextAvailable, setIsTextAvailable] = React.useState(false)
  const [isFocus, setIsFocus] = React.useState(false)

  let { discoverySearchBarText } = generalComponent

  const __handleBackPress = () => {
    if(navigation.canGoBack()) navigation.goBack()
  }

  const __handleFocus = (isFocus) => {
    setIsFocus(isFocus)
  }

  const __handleChangeText = (text) => {
    setIsTextAvailable(text.length > 0)
    GeneralComponentAction.setDiscoverySearchBar(text, generalComponentDispatch)
  }

  const __handleOnClearText = () => {
    __handleChangeText("")
    setIsTextAvailable(false)
    discoverySearchBarRef.current.focus()
  }

  const __fetchDiscoveryData = async() => {
    let data = await DiscoveryRepo.fetchDiscoveryData(discoverySearchBarText)
    if(data.success) {
      DiscoveryAction.setDiscoveryData(data, discoveryDispatch)
    }
    DiscoveryAction.setDiscoveryLoadingData(false, discoveryDispatch)
  }

  React.useEffect(() => {
    if(discoverySearchBarText.length > 1) {
      DiscoveryAction.setDiscoveryLoadingData(true, discoveryDispatch)
      if(getDataTimeoutId) clearTimeout(getDataTimeoutId)
      
      getDataTimeoutId = setTimeout( async() => {
        await __fetchDiscoveryData()
      }, 3000)
    }
  }, [discoverySearchBarText])

  React.useEffect(() => {
    const unsubscribe = () => {
      setIsTextAvailable(false)
      GeneralComponentAction.setDiscoverySearchBar('', generalComponentDispatch)
      DiscoveryAction.setDiscoveryData({
        followedUsers: [], 
        unfollowedUsers: [], 
        followedDomains: [],
        unfollowedDomains: [], 
        followedTopic: [], 
        unfollowedTopic: [], 
        news: [],
      }, discoveryDispatch)
    }

    return unsubscribe
  },[])

  return (
    <Animated.View style={styles.animatedViewContainer(animatedValue)}>
      <Pressable onPress={__handleBackPress}
        android_ripple={{
          color: COLORS.gray1,
          borderless: true,
          radius: 20,
        }} style={styles.arrowContainer}>
        <View style={styles.backArrow}>        
          <MemoIc_arrow_back_white width={20} height={12} fill={colors.black} style={{ alignSelf: 'center'}}/>
        </View>
      </Pressable>
      <Pressable onPress={onContainerClicked} style={styles.searchContainer}>
        <View style={styles.wrapperSearch}>
          {(!isTextAvailable && !isFocus) && <View style={styles.wrapperIcon}>
            <MemoIc_search width={16.67} height={16.67} />
          </View>
          }
          {/* <Text style={styles.inputText}>{StringConstant.newsTabHeaderPlaceholder}</Text> */}
          <TextInput
            ref={discoverySearchBarRef}
            focusable={true}
            autoFocus={true}
            value={generalComponent.discoverySearchBarText}
            onChangeText={__handleChangeText}
            onFocus={() => __handleFocus(true)}
            onBlur={() => __handleFocus(false)}
            multiline={false}
            placeholder={StringConstant.discoverySearchBarPlaceholder}
            placeholderTextColor={COLORS.gray1}
            style={styles.input} />

          {(isTextAvailable || isFocus) && <Pressable onPress={__handleOnClearText} style={styles.clearIconContainer}
            android_ripple={{
              color: COLORS.gray1,
              borderless: true,
              radius: 14,
            }}>
            <View style={styles.wrapperDeleteIcon}>
              <IconClear width={9} height={10} iconColor={colors.black}/>
            </View>
          </Pressable> }
        </View>
      </Pressable>
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  arrowContainer: {paddingLeft:20},
  backArrow : {flex: 1, justifyContent: 'center', marginRight: 9, },
  container: {
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base,
    marginHorizontal: SIZES.base,
  },
  clearIconContainer: {
    justifyContent: 'center', 
    alignItems: 'center',
  },
  searchContainer: {
    flex: 1,
    marginRight: 20,
  },
  wrapperSearch: {
    flex: 1,
    backgroundColor: '#F5F5F5',
    borderRadius: 8,
    alignSelf: 'center',
    flexDirection: 'row',
    height: 36,
    paddingRight: 8,
  },
  wrapperButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginEnd: SIZES.base,
    paddingLeft: 8,
    paddingRight: 8,
    paddingTop: 9,
    paddingBottom: 9,
  },
  input: {
    marginRight: 16,
    paddingStart: 8,
    flex: 1,
    fontSize: 14,
    fontFamily: fonts.inter[400],
    height: 36,
    paddingTop: 0,
    paddingBottom: 0,
  },
  inputText: {
    marginRight: 16,
    paddingStart: 10,
    flex: 1,
    fontSize: 14,
    fontFamily: fonts.inter[400],
    // height: 36,
    paddingTop: 0,
    paddingBottom: 0,
    color: COLORS.gray1,
    alignSelf: 'center'
  },
  wrapperIcon: {
    marginLeft: 9.67,
    marginRight: 1.67,
    alignSelf: 'center',
    justifyContent: 'center',
  },
  wrapperDeleteIcon: {
    alignSelf: 'center',
    justifyContent: 'center',
    marginRight: 9.5,
  },
  newPostText: {
    color: COLORS.holyTosca,
    marginRight: 11,
    ...FONTS.h3,
  },
  animatedViewContainer: (animatedValue) => ({
    flexDirection: 'row',
    backgroundColor: 'white',
    marginTop: animatedValue,
    zIndex: 10,
    height: dimen.size.DISCOVERY_HEADER_HEIGHT,
    paddingTop: 7,
    paddingBottom: 7,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.alto,
    // paddingLeft: 20,
    // paddingRight: 20,
  }),
});

export default DiscoverySearch;
