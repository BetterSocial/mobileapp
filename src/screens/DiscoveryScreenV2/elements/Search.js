/* eslint-disable no-use-before-define */
import * as React from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {
  Keyboard,
  StyleSheet,
  TextInput,
  TouchableNativeFeedback,
  TouchableOpacity,
  View
} from 'react-native';
import {debounce} from 'lodash';
import {useNavigation} from '@react-navigation/core';

import PropTypes from 'prop-types';
import DiscoveryAction from '../../../context/actions/discoveryAction';
import IconClear from '../../../assets/icon/IconClear';
import MemoIcArrowBackWhite from '../../../assets/arrow/Ic_arrow_back_white';
import MemoIcSearch from '../../../assets/icons/Ic_search';
import StringConstant from '../../../utils/string/StringConstant';
import TestIdConstant from '../../../utils/testId';
import dimen from '../../../utils/dimen';
import {COLORS, FONTS, SIZES} from '../../../utils/theme';
import {Context} from '../../../context/Store';
import {RECENT_SEARCH_TERMS} from '../../../utils/cache/constant';
import {colors} from '../../../utils/colors';
import {fonts, normalizeFontSize} from '../../../utils/fonts';

const DiscoverySearch = ({
  setDiscoveryLoadingData = () => {},
  searchText = '',
  setSearchText = () => {},
  placeholderText = StringConstant.discoverySearchBarPlaceholder,
  setIsFocus = () => {},
  setIsFirstTimeOpen = () => {},
  fetchDiscoveryData = () => {},
  fetchData,
  onCancelToken = () => {},
  hideBackIcon = false
}) => {
  const navigation = useNavigation();
  const [, discoveryDispatch] = React.useContext(Context).discovery;
  const discoverySearchBarRef = React.useRef(null);

  const [lastSearch, setLastSearch] = React.useState('');

  const debounced = React.useCallback(
    debounce((text) => {
      handleSubmitSearchData(text);
    }, 1000),

    []
  );

  const handleBackPress = () => {
    Keyboard.dismiss();
    if (navigation.canGoBack()) navigation.goBack();
  };

  const handleFocus = (isFocusParam) => {
    setIsFocus(isFocusParam);
  };

  const setAllLoading = (isLoading) => {
    setDiscoveryLoadingData({
      user: isLoading,
      topic: isLoading,
      domain: isLoading,
      news: isLoading
    });
  };

  const debounceChangeText = (text) => {
    onCancelToken();
    if (text.length > 2) {
      setAllLoading(true);
      setIsFirstTimeOpen(false);
      debounced(text);
    } else {
      setIsFirstTimeOpen(true);
      setAllLoading(false);
      debounced(text);
      debounced.cancel();
    }
  };

  const handleChangeText = (text) => {
    setSearchText(text);
    debounceChangeText(text);
  };

  const handleOnClearText = () => {
    setSearchText('');
    setLastSearch('');
  };

  const handleOnSubmitEditing = (event) => {
    const {text} = event?.nativeEvent;
    handleSubmitSearchData(text);
  };

  const handleSubmitSearchData = async (text) => {
    if (text === lastSearch) return;
    setLastSearch(text);
    setAllLoading(true);
    setIsFirstTimeOpen(false);
    if (fetchData !== undefined) {
      fetchData(true, text);
    } else {
      fetchDiscoveryData(text);
    }

    const result = await AsyncStorage.getItem(RECENT_SEARCH_TERMS);

    if (!result) {
      const itemToSave = JSON.stringify([text]);
      DiscoveryAction.setDiscoveryRecentSearch([text], discoveryDispatch);
      AsyncStorage.setItem(RECENT_SEARCH_TERMS, itemToSave);
      return;
    }

    let resultArray = JSON.parse(result);
    if (resultArray.indexOf(text) > -1) return;

    resultArray = [text].concat(resultArray);
    if (resultArray.length > 3) resultArray.pop();

    DiscoveryAction.setDiscoveryRecentSearch(resultArray, discoveryDispatch);
    AsyncStorage.setItem(RECENT_SEARCH_TERMS, JSON.stringify(resultArray));
  };

  React.useEffect(() => {
    debounceChangeText(searchText);
  }, [searchText]);

  React.useEffect(() => {
    if (discoverySearchBarRef?.current) {
      setTimeout(() => {
        discoverySearchBarRef?.current?.focus();
      }, 500);
    }
  }, []);

  React.useEffect(() => {
    const unsubscribe = () => {
      setSearchText('');
      DiscoveryAction.setDiscoveryData(
        {
          followedUsers: [],
          unfollowedUsers: [],
          followedDomains: [],
          unfollowedDomains: [],
          followedTopic: [],
          unfollowedTopic: [],
          news: []
        },
        discoveryDispatch
      );
    };

    return unsubscribe;
  }, []);

  return (
    <View style={styles.animatedViewContainer(hideBackIcon)}>
      <View style={styles.arrowContainer}>
        {!hideBackIcon && (
          <TouchableNativeFeedback
            testID={TestIdConstant.discoveryScreenBackArrow}
            onPress={handleBackPress}>
            <View style={styles.backArrow}>
              <MemoIcArrowBackWhite
                width={20}
                height={12}
                fill={colors.black}
                style={{alignSelf: 'center'}}
              />
            </View>
          </TouchableNativeFeedback>
        )}
      </View>
      <View style={styles.searchContainer}>
        <View style={styles.wrapperSearch}>
          <View style={styles.wrapperIcon}>
            <MemoIcSearch width={16.67} height={16.67} />
          </View>
          <TextInput
            ref={discoverySearchBarRef}
            testID={TestIdConstant.discoveryScreenSearchBar}
            // value={discoverySearchBarText}
            value={searchText}
            onChangeText={handleChangeText}
            onFocus={() => handleFocus(true)}
            onBlur={() => handleFocus(false)}
            multiline={false}
            returnKeyType="search"
            onSubmitEditing={handleOnSubmitEditing}
            placeholder={placeholderText}
            placeholderTextColor={COLORS.gray1}
            style={styles.input}
          />

          <TouchableOpacity
            testID={TestIdConstant.discoveryScreenClearButton}
            delayPressIn={0}
            onPress={handleOnClearText}
            style={styles.clearIconContainer}
            android_ripple={{
              color: COLORS.gray1,
              borderless: true,
              radius: 35
            }}>
            <View style={styles.wrapperDeleteIcon}>
              <IconClear width={9} height={10} iconColor={colors.black} />
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

DiscoverySearch.propTypes = {
  setDiscoveryLoadingData: PropTypes.func,
  searchText: PropTypes.string,
  setSearchText: PropTypes.func,
  placeholderText: PropTypes.string,
  setIsFocus: PropTypes.func,
  setIsFirstTimeOpen: PropTypes.func,
  fetchDiscoveryData: PropTypes.func,
  fetchData: PropTypes.func,
  onCancelToken: PropTypes.func,
  hideBackIcon: PropTypes.bool
};

const styles = StyleSheet.create({
  arrowContainer: {paddingLeft: 20},
  backArrow: {
    flex: 1,
    justifyContent: 'center',
    marginRight: dimen.normalizeDimen(9)
  },
  container: {
    flexDirection: 'row',
    backgroundColor: COLORS.white,
    marginBottom: SIZES.base,
    marginHorizontal: SIZES.base
  },
  clearIconContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: dimen.normalizeDimen(-20.5),
    paddingHorizontal: dimen.normalizeDimen(30),
    zIndex: 1000
  },
  searchContainer: {
    flex: 1,
    marginRight: dimen.normalizeDimen(20)
  },
  wrapperSearch: {
    flex: 1,
    backgroundColor: COLORS.lightgrey,
    borderRadius: dimen.normalizeDimen(8),
    alignSelf: 'center',
    flexDirection: 'row',
    height: dimen.normalizeDimen(36),
    paddingRight: dimen.normalizeDimen(8)
  },
  wrapperButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginEnd: SIZES.base,
    paddingHorizontal: dimen.normalizeDimen(8),
    paddingVertical: dimen.normalizeDimen(9)
  },
  input: {
    marginRight: dimen.normalizeDimen(16),
    paddingStart: dimen.normalizeDimen(8),
    flex: 1,
    fontSize: normalizeFontSize(14),
    fontFamily: fonts.inter[400],
    alignSelf: 'center',
    height: dimen.normalizeDimen(33),
    paddingTop: 0,
    paddingBottom: 0
  },
  wrapperIcon: {
    marginLeft: dimen.normalizeDimen(9.67),
    marginRight: dimen.normalizeDimen(1.67),
    alignSelf: 'center',
    justifyContent: 'center'
  },
  wrapperDeleteIcon: {
    alignSelf: 'center',
    justifyContent: 'center'
  },
  newPostText: {
    color: COLORS.holytosca,
    marginRight: dimen.normalizeDimen(11),
    ...FONTS.h3
  },
  animatedViewContainer: (hideBackIcon) => ({
    flexDirection: 'row',
    backgroundColor: 'white',
    marginTop: 0,
    zIndex: 10,
    height: dimen.size.DISCOVERY_HEADER_HEIGHT,
    paddingVertical: dimen.normalizeDimen(7),
    borderBottomWidth: hideBackIcon ? 0 : dimen.normalizeDimen(1),
    borderBottomColor: COLORS.alto
  })
});

DiscoverySearch.propTypes = {
  setDiscoveryLoadingData: PropTypes.func,
  searchText: PropTypes.string,
  setSearchText: PropTypes.func,
  placeholderText: PropTypes.string,
  setIsFocus: PropTypes.func,
  setIsFirstTimeOpen: PropTypes.func,
  fetchDiscoveryData: PropTypes.func,
  fetchData: PropTypes.func,
  onCancelToken: PropTypes.func,
  hideBackIcon: PropTypes.bool
};

export default DiscoverySearch;
