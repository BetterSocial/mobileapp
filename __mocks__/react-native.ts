import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN