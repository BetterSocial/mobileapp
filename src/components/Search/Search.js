import * as React from 'react';
import {TextInput, View, StyleSheet} from 'react-native';

import SearchIcon from '../../../assets/icons/search.svg';

const Search = (props) => {
  return (
    <View style={{...styles.inputContainer, ...styles.container}}>
      <SearchIcon width={16.67} height={16.67} fill="#000" />
      <TextInput
        style={{...styles.searchInput, ...styles.textInput}}
        onChangeText={props.onChangeText}
        value={props.value}
        placeholder={props.placeholder ? props.placeholder : ''}
      />
    </View>
  );
};
const styles = StyleSheet.create({
  inputContainer: {
    height: 48,
    backgroundColor: '#FFFFFF',
    borderWidth: 1,
    borderColor: '#BDBDBD',
    borderRadius: 8,
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 13,
  },
  searchInput: {
    width: '90%',
    height: 40,
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 13,
    letterSpacing: -0.28,
    color: '#BDBDBD',
    marginLeft: 5,
  },
});
export default Search;
