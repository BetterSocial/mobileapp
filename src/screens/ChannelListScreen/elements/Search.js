/* eslint-disable no-use-before-define */
/* eslint-disable no-underscore-dangle */
import * as React from 'react';
import PropTypes from 'prop-types';
import {Pressable, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {useNavigation} from '@react-navigation/core';

import MemoIcNewChat from '../../../assets/icons/ic_new_chat';
import MemoIcSearch from '../../../assets/icons/Ic_search';
import StringConstant from '../../../utils/string/StringConstant';
import dimen from '../../../utils/dimen';
import {DISCOVERY_TAB_USERS} from '../../../utils/constants';
import {SIZES} from '../../../utils/theme';
import {colors} from '../../../utils/colors';
import {fonts} from '../../../utils/fonts';

const Search = ({route, onPress, isShowNewChat = true}) => {
  const navigation = useNavigation();

  const __handleOnSearchClicked = () => {
    navigation.push('DiscoveryScreen', {
      tab: DISCOVERY_TAB_USERS
    });
  };

  const themeColor = () => {
    if (route?.name === 'SignedChannelList') return colors.darkBlue;
    return colors.anon_primary;
  };

  return (
    <View style={styles.animatedViewContainer}>
      <Pressable onPress={__handleOnSearchClicked} style={styles.searchPressableContainer}>
        <View style={styles.wrapperSearch}>
          <View style={styles.wrapperIcon}>
            <MemoIcSearch width={16.67} height={16.67} />
          </View>
          <Text style={styles.input}>{StringConstant.chatTabHeaderPlaceholder}</Text>
        </View>
      </Pressable>
      {isShowNewChat && (
        <TouchableOpacity style={styles.wrapperButton} onPress={onPress}>
          <Text style={[styles.newPostText, {color: themeColor()}]}>
            {StringConstant.chatTabHeaderCreateChatButtonText}
          </Text>
          <View>
            <MemoIcNewChat height={18} width={16} color={themeColor()} style={styles.newChatIcon} />
          </View>
        </TouchableOpacity>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base
  },
  searchPressableContainer: {
    flex: 1
  },
  wrapperSearch: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: '#F5F5F5',
    marginLeft: 20,
    marginRight: 12,
    borderRadius: 8,
    alignSelf: 'center',
    height: 36
  },
  wrapperButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginEnd: SIZES.base,
    paddingRight: 12,
    paddingTop: 9,
    paddingBottom: 9
  },
  input: {
    paddingStart: 10,
    lineHeight: 36,
    flex: 1,
    paddingBottom: 0,
    paddingTop: 0,
    fontFamily: fonts.inter[400],
    fontSize: 14,
    alignSelf: 'center',
    color: colors.gray1
  },
  wrapperIcon: {
    marginLeft: 8,
    alignSelf: 'center',
    justifyContent: 'center'
  },
  newPostText: {
    marginRight: dimen.normalizeDimen(5),
    fontFamily: 'Inter-SemiBold',
    fontSize: 12,
    lineHeight: 14.52
  },
  newChatIcon: {
    marginTop: 0
  },
  animatedViewContainer: {
    flexDirection: 'row',
    backgroundColor: 'white',
    marginBottom: SIZES.base,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    zIndex: 10,
    paddingTop: 7,
    paddingBottom: 7,
    borderBottomWidth: 1,
    borderBottomColor: colors.alto
  }
});

Search.propTypes = {
  route: PropTypes.object.isRequired,
  onPress: PropTypes.func.isRequired
};

export default Search;
