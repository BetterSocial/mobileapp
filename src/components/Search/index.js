import Search from './Search';
import SearchAutoComplete from './SearchAutoComplete';
import SearchModal from './SearchModal';

export {Search, SearchAutoComplete, SearchModal};
