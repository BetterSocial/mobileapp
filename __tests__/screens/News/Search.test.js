import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react-native';

import Search from '../../../src/screens/NewsScreen/Search';

jest.mock('react-native/Libraries/Pressability/usePressability');
const mockPush = jest.fn();
const mockBack = jest.fn();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({goBack: mockBack, push: mockPush}),
  useRoute: () => ({
    params: {}
  })
}));
describe('News search should run correctly', () => {
  afterEach(cleanup);
  it('Should match snapshot', () => {
    const {toJSON} = render(<Search />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('containerPress should run correctly', () => {
    const {getByTestId} = render(<Search />);
    fireEvent.press(getByTestId('containerPress'));
    expect(mockPush).toHaveBeenCalled();
  });

  it('arrow left should navigate back', () => {
    const {getByTestId} = render(<Search />);
    fireEvent.press(getByTestId('news-search-back-button'));
    expect(mockBack).toHaveBeenCalled();
  });
});
