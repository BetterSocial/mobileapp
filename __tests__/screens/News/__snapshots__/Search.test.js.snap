// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News search should run correctly Should match snapshot 1`] = `[Function]`;
